fastestimator/pipeline.py

```diff
@@ -181,81 +215,84 @@ def benchmark(self,
-                                unique_samples.add(id(item))
-                    else:
-                        for i, op in enumerate(op_list):
-                            start = time.perf_counter()
-                            forward_numpyop([op], items, {'mode': loader.dataset.mode})
-                            duration = time.perf_counter() - start
-                            duration_list[i] += duration
-
-                total_time = np.sum(duration_list)
-                op_names = ["Op"]
-
-                for op in op_list:
-                    if isinstance(op, Sometimes) and op.op:
-                        op_names.append(op.__class__.__name__ + " (" + op.op.__class__.__name__ + ")")
-                    elif isinstance(op, OneOf) and op.ops:
-                        op_names.append(op.__class__.__name__ + " (" +
-                                        ", ".join([sub_op.__class__.__name__ for sub_op in op.ops]) + ")")
-                    else:
-                        op_names.append(op.__class__.__name__)
-
-                max_op_len = max(len(op_name) for op_name in op_names)
-                max_in_len = max([len(", ".join(op.inputs)) for op in op_list] + [len("Inputs")])
-                max_out_len = max([len(", ".join(op.outputs)) for op in op_list] + [len("Outputs")])
-                print("{}: {}: {}: {}".format("Op".ljust(max_op_len + 1),
-                                              "Inputs".ljust(max_in_len + 1),
-                                              "Outputs".ljust(max_out_len + 1),
-                                              "Time".rjust(5)))
-                print("-" * (max_op_len + max_in_len + max_out_len + 15))
-                for i, op in enumerate(op_list):
-                    print("{}: {}: {}: {:5.2f}%".format(op_names[i + 1].ljust(max_op_len + 1),
-                                                        ", ".join(op.inputs).ljust(max_in_len + 1),
-                                                        ", ".join(op.outputs).ljust(max_out_len + 1),
-                                                        100 * duration_list[i] / total_time))
+        if ds_id is None:
+            ds_ids = self.get_ds_ids(epoch=epoch, mode=mode)
+        else:
+            ds_ids = [ds_id]
+        idx = 0
+        for ds_id in ds_ids:
+            if idx >= num_steps:
+                break
+            with self(mode=mode, epoch=epoch, ds_id=ds_id) as loader:
+                if isinstance(loader, tf.data.Dataset):
+                    loader = loader.take(num_steps)
+                start = time.perf_counter()
+                for idx, _ in enumerate(loader, start=idx + 1):
+                    if idx % log_interval == 0:
+                        duration = time.perf_counter() - start
+                        iters_per_sec = log_interval / duration
+                        print("FastEstimator: Step: {}, Epoch: {}, Steps/sec: {}".format(idx, epoch, iters_per_sec))
```

Comment:
seems like we should report ds_id here or somewhere as well when it's not None

Reply:
reporting ds_id, the benchmark behavior is changed too.